src/ZoneServer/Scripting/AI/AiScript.cs

```diff
@@ -626,6 +629,27 @@ protected bool EntityGone(ICombatEntity entity)
 			return false;
 		}
 
+		/// <summary>
+		/// Gets a valid position adjacent to target within given range.
+		/// Returns entity's own position if not found.
+		/// </summary>
+		/// <param name="entity"></param>
+		/// <returns></returns>
+		protected Position GetAdjacentValidPosition(ICombatEntity entity, float range)
+		{
+			var ground = entity.Map.Ground;
+
+			for (var i = 0; i < 10; i++)
```

Comment:
Just one more question, do we really think that monsters choose an entirely random position _somewhere_ around their target? From what I've seen, they absolutely do choose an adjacent position at times, but to me it seemed to always be between the attacker and the target, and not like they would potentially decide to pass you entirely and attack from behind. If that happened on officials, I would rather assume that the AI overshot their target accidentally 🤔

Reply:
I'm currently writing an alternative to this function, but for reference I wanted to mention that I think there's another potential issue with this method as the source for attack positions, in that monsters could randomly run back and forth as the player moves, alternating attacking them from the front or back, because a random position is chosen every time the player moves at least 10 units. It would probably not happen a lot in practice, but it might look weird when it does happen.